src/base/notifications/notificationoption.cpp

```diff
@@ -0,0 +1,132 @@
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "notificationoption.h"
+
+Notifications::NotificationOptionDesc::NotificationOptionDesc(const std::string &id, const QString &name)
+    : m_id {id}
+    , m_name {name}
+    , m_enabledByDefault {false}
+    , m_signalSource {nullptr}
+    , m_receiver {nullptr}
+
+{
+}
+
+Notifications::NotificationOptionDesc::ThisType &Notifications::NotificationOptionDesc::setDescription(const QString &v)
+{
+    m_description = v;
+    return *this;
+}
+
+Notifications::NotificationOptionDesc::ThisType &Notifications::NotificationOptionDesc::setEnabledByDefault(bool v)
+{
+    m_enabledByDefault = v;
+    return *this;
+}
+
+Notifications::NotificationOptionDesc::ThisType
+&Notifications::NotificationOptionDesc::setSignalConnection(QObject *signalSource, const std::string &signalName,
+                                                            QObject *receiver, const std::string &slotName)
+{
+    m_signalSource = signalSource;
+    m_signalName = signalName;
+    m_receiver = receiver;
+    m_slotName = slotName;
+
+    return *this;
+}
+
+const std::string &Notifications::NotificationOptionDesc::id() const
+{
+    return m_id;
+}
+
+const QString &Notifications::NotificationOptionDesc::name() const
+{
+    return m_name;
+}
+
+const QString &Notifications::NotificationOptionDesc::description() const
+{
+    return m_description;
+}
+
+bool Notifications::NotificationOptionDesc::isEnabledByDefault() const
+{
+    return m_enabledByDefault;
+}
+
+void Notifications::NotificationOptionDesc::assertValidity() const
+{
+    Q_ASSERT(!m_id.empty());
+    Q_ASSERT(!m_name.isEmpty());
+    Q_ASSERT(!m_description.isEmpty());
+    Q_ASSERT(m_signalSource != nullptr);
+    Q_ASSERT(!m_signalName.empty());
+    Q_ASSERT(m_receiver != nullptr);
+    Q_ASSERT(!m_slotName.empty());
+}
+
+bool Notifications::operator==(const Notifications::NotificationOptionDesc &left, const Notifications::NotificationOptionDesc &right)
+{
+    return left.id() == right.id();
+}
+
+Notifications::NotificationOption::NotificationOption(const NotificationOptionDesc &desc)
+    : base {desc}
+    , m_isConnectionEstablished {false}
+{
+    assertValidity();
+}
+
+Notifications::NotificationOption::~NotificationOption()
+{
+    setActive(false);
+}
+
+bool Notifications::NotificationOption::isActive() const
+{
+    return m_isConnectionEstablished;
+}
+
+void Notifications::NotificationOption::setActive(bool active)
+{
+    if (active && !isActive()) {
+        Q_ASSERT(QObject::connect(m_signalSource, m_signalName.c_str(), m_receiver, m_slotName.c_str()));
```

Comment:
Can understand your point. I need a check that connection succeeds. The check works.